### DISCUSSION TITLE
Logging in validation step that is the same as training step

### DISCUSSION THREAD

**Discussion by RafailFridman** [Lightning-AI/pytorch-lightning#9334] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hi, my validation step is the same as the training step:
```python
def validation_step(self, batch, batch_idx):
    return self.training_step(batch, batch_idx)
```
In training step I call logging of some metrics like this:
```python 
self.log("train/combined_loss", loss, prog_bar=False, on_epoch=True, on_step=False)
```

But what I want to do is replace `"train/combined_loss"` with something like `f"{self.current_stage}/combined_loss"`
Is it possible to achieve somehow?

**Comment by RafailFridman** [comment]
I'm now doing like this:
```python
def training_step(self, batch, batch_idx):
    logging_prefix = "train" if self.trainer.training else "val"
    ...
    self.log(f"{logging_prefix}/combined_loss", loss, prog_bar=False, on_epoch=True, on_step=False)
```
But maybe there is a better way?

**Reply by rohitgr7** [reply]
this looks ok.. you can also use `self.log(f"{self.trainer.state.stage}/combined_loss"` to avoid that one line of code.

Also here:
```py
def validation_step(self, batch, batch_idx):
    return self.training_step(batch, batch_idx)
```
I'd suggest creating another custom function which is called within both `training_step` and `validation_step`.
